docs/user/machine-api-operator-overview.md

```diff
@@ -0,0 +1,85 @@
+# Document Purpose
+
+This document describes how the *Machine API Operator* (herein MAO) gets deployed, and the MAO's responsibilities. It includes guidelines on how to troubleshoot typical issues with resources provisioned by MAO.
+
+## CVO integration
+
+MAO code is running under the `machine-api-operator` Deployment. This and other resources listed below rely on the *Cluster Version Operator* (herein CVO) management strategy". These resources are consumed from the ["/install" directory]( https://github.com/openshift/machine-api-operator/tree/master/install).
+
+Namely:
+- `openshift-machine-api` Namespace - the primary and the only namespace, where the MAO is hosted, and all of the dependent resources are being managed by the operator.
+- `machine-api-operator` Deployment - MAO code runs here.
+- `machine-api-operator-images` ConfigMap - is injected with provider images, used by MAO to deploy cloud-provider specific code.
+- CredentialsRequests - granular request for cloud credentials, fulfilled by the [cloud-credential-operator](https://github.com/openshift/cloud-credential-operator)
+- ImageStreams for the ConfigMap with provider image mapping - a set of plain DockerImages, injected into OCP release payload.
+- Alerts, Metrics, CRDs, RBAC, and Services are deployed from the "/install" directory.
+
+### Upgrade
+
+MAO installation/upgrade procedure relies on CVO. 
+
+MAO is deployed after CVO starts running at [run level](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/operators.md#how-do-i-get-added-as-a-special-run-level) 30, right after the Kubernetes operators (run level 10-29), and before the `machine-config-operator` (run level 80)
+
+The new version of MAO Deployment is responsible for installing/upgrading provider executables
```

Comment:
the "new version" part here is a little confusing to me, what is this referring to?

Reply:
What I wanted to say "rolled-out version of MAO Deployment is ..." as trying to describe upgrade order or procedure. Good question.

Reply:
maybe could say "A newly deployed version of MAO is responsible for ...."?

Reply:
Done